From 872581355f4bff115abd849267b52bb96ef8d4bb Mon Sep 17 00:00:00 2001
From: Ben Cressey <bcressey@amazon.com>
Date: Mon, 8 Nov 2021 05:13:35 +0000
Subject: [PATCH 4/4] adjust safeguard for dhcp6 defer timeout

For low timeout values, such as one second, we could sometimes wait
twice as long as expected if the timer fired early. If we're within
one second of the deadline, consider it done.

Signed-off-by: Ben Cressey <bcressey@amazon.com>
---
 src/dhcp6/fsm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/dhcp6/fsm.c b/src/dhcp6/fsm.c
index 08624b49..b864efd6 100644
--- a/src/dhcp6/fsm.c
+++ b/src/dhcp6/fsm.c
@@ -569,7 +569,7 @@ ni_dhcp6_fsm_timeout(ni_dhcp6_device_t *dev)
 			/* Do we still need this safeguard? */
 			deadline = ni_lifetime_left(dev->config->defer_timeout,
 					&dev->start_time, NULL);
-			if (deadline) {
+			if (deadline > 1) {
 				ni_dhcp6_fsm_set_timeout_sec(dev, deadline);
 				dev->fsm.fail_on_timeout = 0;
 				return;
-- 
2.32.0

